core/trino-spi/src/main/java/io/trino/spi/exchange/Exchange.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.exchange;
+
+import io.airlift.slice.Slice;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+@ThreadSafe
+public interface Exchange
+        extends Closeable
+{
+    /**
+     * Registers a new sink
+     *
+     * @param taskPartitionId uniquely identifies a dataset written to a sink
+     * @return {@link ExchangeSinkHandle} associated with the <code>taskPartitionId</code>.
+     * Must be passed to {@link #instantiateSink(ExchangeSinkHandle, int)} along with an attempt id to create a sink instance
```

Comment:
nit: I think ti should be before `@param..@return` section per convention

Reply:
Or is that a part of `@return` description?

Reply:
This is the continuation of the `@return` description